src/app/reporting/Engine.cpp

```diff
@@ -104,28 +99,44 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeDataList(ReportData::Builder &
     aReportDataBuilder.Checkpoint(backup);
     AttributeDataList::Builder attributeDataList = aReportDataBuilder.CreateAttributeDataListBuilder();
     SuccessOrExit(err = aReportDataBuilder.GetError());
-    // TODO: Need to handle multiple chunk of message
-    for (auto clusterInfo = apReadHandler->GetAttributeClusterInfolist(); clusterInfo != nullptr; clusterInfo = clusterInfo->mpNext)
+
+    if (apReadHandler->IsInitialReport())
     {
-        if (apReadHandler->IsInitialReport())
+        ConcreteAttributePath path;
+        mMoreChunkedMessages = true;
+        for (; apReadHandler->GetPathIterator()->Get(path); apReadHandler->GetPathIterator()->Proceed())
         {
+            TLV::TLVWriter attributeBackup;
+            aReportDataBuilder.Checkpoint(attributeBackup);
             // Retrieve data for this cluster instance and clear its dirty flag.
-            err = RetrieveClusterData(apReadHandler->GetFabricIndex(), attributeDataList, *clusterInfo);
+            err = RetrieveClusterData(apReadHandler->GetFabricIndex(), attributeDataList, path);
+            if (err != CHIP_NO_ERROR)
+            {
+                aReportDataBuilder.Rollback(attributeBackup);
+            }
             VerifyOrExit(err == CHIP_NO_ERROR,
                          ChipLogError(DataManagement, "<RE:Run> Error retrieving data from cluster, aborting"));
```

Comment:
Please don't print this out in the case where you ran out of space - it's mis-leading to say you're aborting when this is actually a completely reasonable error condition that is recoverable.